### DISCUSSION TITLE
How to generate sql migration file?

### DISCUSSION THREAD

**Discussion by valerii15298** [ent/ent#3835] [Q&A] [answered]
I had read docs that it is possible to use atlas.
But can we create migration sql file without it?
For example, ent has auto migration function which creates db tables.
Can I somehow create that sql file which ent uses to create tables from schema in empty database?
Similar how prisma migrate create only does(https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#customizing-migrations)

**Comment by keenanwl** [comment, marked as answer]
Yes, you can. Sounds like you are looking for [Offline Mode](https://entgo.io/docs/migrate#offline-mode)?

AFAIK from v0.11, Atlas is used under the hood.
